### DISCUSSION TITLE
How to do CI/CD for hybrid SSG/SSR?

### DISCUSSION THREAD

**Discussion by Evian-Zhang** [vercel/next.js#12585] [Help]
I am developing a website using NextJS with both SSG and SSR. My SSG needs access to my server's internal net, which does not export to others. So how can I do CI/CD with my project? Maybe next cli can export a `compile` option for test CI, which does not build the page and only tests if the project compiles successfully.

**Comment by Vadorequest** [comment]
I use Github Actions. See https://github.com/UnlyEd/next-right-now

More specifically:
- https://unlyed.github.io/next-right-now/concepts/ci-cd
- https://unlyed.github.io/next-right-now/guides/ci-cd/
- https://github.com/UnlyEd/next-right-now/tree/v1-hyb-mst-aptd-gcms-lcz-sty/.github/workflows
